### PR TITLE
fix: [CI-3363]: fix k8 inputset handling

### PR DESCRIPTION
**Pull request by harshjain12**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30534)

### CODE REVIEW

File: 320-ci-execution/src/main/java/io/harness/states/CleanupStep.java [hunk 1]
```diff
@@ -78,8 +78,8 @@ public TaskRequest obtainTask(Ambiance ambiance, CleanupStepInfo stepParameters,
     // It should always resolved to K8sDirectInfraYaml
     K8sDirectInfraYaml k8sDirectInfraYaml = (K8sDirectInfraYaml) infrastructure;
 
-    final String clusterName = k8sDirectInfraYaml.getSpec().getConnectorRef();
-    final String namespace = k8sDirectInfraYaml.getSpec().getNamespace();
+    final String clusterName = (String) k8sDirectInfraYaml.getSpec().getConnectorRef().fetchFinalValue();
```
Comment (shubham149) [inline 320-ci-execution/src/main/java/io/harness/states/CleanupStep.java:81]: Do we have to check for null values k8sDirectInfraYaml.getSpec().getConnectorRef() or k8sDirectInfraYaml.getSpec().getNamespace()

Comment (harshjain12) [reply 320-ci-execution/src/main/java/io/harness/states/CleanupStep.java:81]: No, we don't have to because of not null annotation in yaml

**Review by shubham149 — APPROVED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30534)* status: :shipit: complete! ✓

**Comment by harshjain12** [comment]
trigger messagecheck

**Pull request merged**